### DISCUSSION TITLE
ForceSay feedback

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#15200] [Unstable feedback]
In 1.7 we have the ForceSay feature https://github.com/FakeFishGames/Barotrauma/pull/14520

I noticed 2 shortcomings:

**1. You need 5 lines to use it in events**
You need a statuseffectaction, statuseffect, and forcesay inside
```xml
      <StatusEffectAction targettag="target">
          <StatusEffect target="this">
            <Forcesay message="eventtext.forcesayTEST.testing" sayinradio="true" />
          </StatusEffect>
      </StatusEffectAction>
    </ScriptedEvent>
```

Having a `ForceSayAction` for events with targettag would reduce that to one line. 
This is where it would shine; having actual conversations in speech bubbles for anyone to see, but the current implementation is cumbersome.
Also potentially implementing a forcesay parameter to ConversationAction

**2. Items can't speak**
Only characters can speak. There's no particular reason why you wouldnt be able to generate speech bubbles on items (wifi components can speak in chat).
Alarms, speakers, armors, sentient axes... 
![image](https://github.com/user-attachments/assets/a8b28f35-4c07-40f0-9b13-383ff0b1b0d3)

**Comment by jigardi** [comment]
Have my upvote!
With a little love this could also be used to make a rudimentary intercom system:
*SpeakerActivationStatic* "This is the captain speaking. All hands, brace for impact!"

**Comment by ubertpendragon** [comment]
Another very useful feature would be to have Forcesay INSIDE Conversationaction, so that you can trigger a speech bubble above the NPC at same time you trigger conversationaction

example:
```xml
<ConversationAction text="The captain says: Hello" forcesay="Hello" ... />
```

**Comment by Regalis11** [comment]
This has now been addressed in our private repo, coming up in the next unstable build and the upcoming spring update.
